src/common/state/slices/modal.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { MODALS } from '@common/constants';
+import { useAppSelector } from '@common/state/hooks';
+
+export interface State {
+  openedModal?: MODALS;
+}
+
+const initialState: State = {
+  openedModal: undefined,
+};
+
+export const slice = createSlice({
+  name: 'modal',
+  initialState,
+  reducers: {
+    openModal: (state, action: PayloadAction<MODALS>) => {
+      state.openedModal = action.payload;
+    },
+    closeModal: state => {
+      state.openedModal = undefined;
+    },
+  },
+});
+
+export const { openModal, closeModal } = slice.actions;
+
+export const getOpenedModal = () => useAppSelector(state => state.modal.openedModal);
```

Comment:
Should we figure out a naming convention for selectors? I've been using `select` Prefixed naming.

Reply:
I'm good with both, I used `get` as it's more framework-agnostic (not associated with reselect) but I think both work well.

Reply:
@beguene @fbwoolf for input

I don't really care either way, but let's just pick one.

Reply:
I would just go with `select` if it is already implemented in the wallet changes. It makes sense to me as prefix.

Reply:
I prefer ```select```.